src/rudderstack.ts

```diff
@@ -37,10 +49,13 @@ export class RudderStack {
      */
     init = (RUDDERSTACK_KEY: string) => {
         if (RUDDERSTACK_KEY) {
-            this.analytics.load(RUDDERSTACK_KEY, 'https://deriv-dataplane.rudderstack.com')
+            this.setCookieIfNotExists()
+            this.analytics.load(RUDDERSTACK_KEY, 'https://deriv-dataplane.rudderstack.com', {
+                externalAnonymousIdCookieName: this.rudderstack_anonymous_cookie_key,
+            })
             this.analytics.ready(() => {
                 this.has_initialized = true
-                this.has_identified = !!(this.getUserId() || !!this.getAnonymousId())
+                this.has_identified = !!(this.getUserId() || this.getAnonymousId())
```

Comment:
[Q] in what case we need this.has_identified?, seems we can even get this value before .ready()

Reply:
getUserId will be available only on ready function.